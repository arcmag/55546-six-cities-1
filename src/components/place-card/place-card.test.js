import renderer from 'react-test-renderer';

import PlaceCard from './place-card';

const dataMock = {
  title: `Beautiful & luxurious apartment at great location`,
  src: `img/apartment-01.jpg`,
  price: 120,
  rating: 4.7,
  type: `Apartment`,
  isChecked: false,
  isPremium: true
};

describe(`Test PlaceCard`, () => {
  it(`Test PlaceCard renderer`, () => {
    const tree = renderer.create(<PlaceCard
      data={dataMock}
      onImgClick={jest.fn()}
      onCardMouseOver={jest.fn()}
      onCardMouseOut={jest.fn()}
    />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
