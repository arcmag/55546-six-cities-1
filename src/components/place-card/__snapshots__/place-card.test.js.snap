// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test PlaceCard Test PlaceCard renderer 1`] = `
<article
  className="cities__place-card place-card"
>
  
  <div
    className="cities__image-wrapper place-card__image-wrapper"
  >
    <a
      href="#"
    >
      <img
        className="place-card__image"
        height="200"
        src=""
        width="260"
      />
    </a>
  </div>
  <div
    className="place-card__info"
  >
    <div
      className="place-card__price-wrapper"
    >
      <div
        className="place-card__price"
      >
        <b
          className="place-card__price-value"
        >
          €
          0
        </b>
        <span
          className="place-card__price-text"
        >
          / night
        </span>
      </div>
      
    </div>
    <div
      className="place-card__rating rating"
    >
      <div
        className="place-card__stars rating__stars"
      >
        <span
          style={
            Object {
              "width": "100%",
            }
          }
        />
        <span
          className="visually-hidden"
        >
          Rating
        </span>
      </div>
    </div>
    <h2
      className="place-card__name"
    >
      <a
        className="header__logo-link"
        href="/offer/1"
        onClick={[Function]}
      >
        
      </a>
    </h2>
    <p
      className="place-card__type"
    >
      
    </p>
  </div>
</article>
`;
